src/AutoComplete/AutoComplete.js

```diff
@@ -265,21 +264,29 @@ class AutoComplete extends Component {
 
     const index = parseInt(child.key, 10);
     const chosenRequest = dataSource[index];
-    const searchText = this.chosenRequestText(chosenRequest);
-
-    this.setState({
-      searchText: searchText,
-    }, () => {
-      this.props.onUpdateInput(searchText, this.props.dataSource, {
-        source: 'touchTap',
-      });
-
+    if (typeof this.props.searchText !== 'undefined') {
       this.timerTouchTapCloseId = setTimeout(() => {
         this.timerTouchTapCloseId = null;
         this.close();
         this.props.onNewRequest(chosenRequest, index);
       }, this.props.menuCloseDelay);
-    });
+    } else {
+      const searchText = this.chosenRequestText(chosenRequest);
+
+      this.setState({
+        searchText: searchText,
+      }, () => {
+        this.props.onUpdateInput(searchText, this.props.dataSource, {
```

Comment:
Sure but that's a breaking change with the current behavior. Users can use the options argument to filter out that call.
I think that we shouldn't call it here but not introducing a breaking change is even more important.